### PR TITLE
bootloader: bl_validation: Fix looping through public keys

### PR DESCRIPTION
**Pull request by oyvindronningstad**
And fix some printing in the same file

### CODE REVIEW

**Comment by oyvindronningstad** [comment]
Added backport label. This fixes NCSDK-3510

File: tests/subsys/bootloader/bl_validation/prj.conf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+CONFIG_ZTEST=y
+CONFIG_ZTEST_STACKSIZE=12384
```
Comment (hakonfam) [inline tests/subsys/bootloader/bl_validation/prj.conf]: Please check if this is really needed.

File: tests/subsys/bootloader/bl_validation/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+tests:
+  bootloader.bl_validation:
+    platform_whitelist: nrf52840_pca10056 nrf52_pca10040 nrf51_pca10028
```
Comment (hakonfam) [inline tests/subsys/bootloader/bl_validation/testcase.yaml]: not 10090?

File: tests/subsys/bootloader/bl_validation/prj.conf (hunk 1 above)
Comment (oyvindronningstad) [reply tests/subsys/bootloader/bl_validation/prj.conf]: This will be needed in the future. Will remove here and add when needed.

Comment (oyvindronningstad) [reply tests/subsys/bootloader/bl_validation/prj.conf]: Done

File: tests/subsys/bootloader/bl_validation/testcase.yaml (hunk 2 above)
Comment (oyvindronningstad) [reply tests/subsys/bootloader/bl_validation/testcase.yaml]: Added

File: subsys/bootloader/image/CMakeLists.txt [hunk 3]
```diff
@@ -51,6 +51,12 @@ set(PROVISION_HEX          ${PROJECT_BINARY_DIR}/${PROVISION_HEX_NAME})
 include(${CMAKE_CURRENT_LIST_DIR}/../cmake/debug_keys.cmake)
 include(${CMAKE_CURRENT_LIST_DIR}/../cmake/sign.cmake)
 
+if (${CONFIG_SB_DEBUG_SIGNATURE_PUBLIC_KEY_LAST})
+  set(ALL_PUBLIC_KEY_FILES "${PUBLIC_KEY_FILES},${SIGNATURE_PUBLIC_KEY_FILE}")
```
Comment (hakonfam) [inline subsys/bootloader/image/CMakeLists.txt]: Please print a big fat warning here. It is really bad if this is enabled by mistake.

File: subsys/bootloader/bl_validation/bl_validation.c [hunk 4]
```diff
@@ -155,25 +155,27 @@ static bool validate_firmware(u32_t fw_dst_address, u32_t fw_src_address,
 		}
 
 		PRINT("Verifying signature against key %d.\n\r", key_data_idx);
+		PRINT("Hash: 0x%02x...0x%02x\r\n", key_data[0] & 0xFF,
```
Comment (hakonfam) [inline subsys/bootloader/bl_validation/bl_validation.c]:
```suggestion
		PRINT("Hash: 0x%02x...%02x\r\n", key_data[0] & 0xFF,
```

File: subsys/bootloader/image/CMakeLists.txt (hunk 3 above)
Comment (oyvindronningstad) [reply subsys/bootloader/image/CMakeLists.txt]: Done

File: subsys/bootloader/bl_validation/bl_validation.c (hunk 4 above)
Comment (oyvindronningstad) [reply subsys/bootloader/bl_validation/bl_validation.c]: Done

**Pull request merged**